QModManager/HarmonyPatches/QuitToDesktop.cs

```diff
@@ -0,0 +1,120 @@
+            quitConfirmation2.GetComponentsInChildren<Button>()[1].onClick.RemoveAllListeners();
+            quitConfirmation2.GetComponentsInChildren<Button>()[1].onClick.AddListener(() => __instance.QuitGame(true));
+        }
+
+        internal static void QuitDesktopSubscreen(IngameMenu __instance)
+        {
+            float time = Time.timeSinceLevelLoad - __instance.lastSavedStateTime;
+            if (!GameModeUtils.IsPermadeath() && time > __instance.maxSecondsToBeRecentlySaved)
+            {
+                // We can't use Language.main directly since it's a field in Subnautica and a property in Below Zero
+                Language languageMain;
+                try
+                {
+                    // We can't use nameof(Language.main) since it will throw a field not found exception in Below Zero
+                    languageMain = AccessTools.Field(typeof(Language), "main").GetValue(null) as Language;
+                }
+                catch
+                {
+                    try
+                    {
+                        languageMain = AccessTools.Property(typeof(Language), "main").GetValue(null, null) as Language;
+                    }
+                    catch
+                    {
+                        quitButton.GetComponentsInChildren<Text>().Do(t => t.text = "ERROR");
+                        quitButton.interactable = false;
+                        return;
+                    }
+                }
+
+                quitConfirmation2.GetComponentsInChildren<Text>()[1].text = languageMain.GetFormat("TimeSinceLastSave", Utils.PrettifyTime((int)time));
+                __instance.ChangeSubscreen("QuitToDesktop ConfirmationWithSaveWarning");
+                return;
+            }
+            __instance.ChangeSubscreen("QuitToDesktop Confirmation");
+        }
+    }
+
+    [HarmonyPatch(typeof(IngameMenu), nameof(IngameMenu.Open))]
+    internal static class IngameMenu_Open_Patch
+    {
+        // This patch disables the QTD button in hardcore mode and changes the names of the two quit buttons based on game mode
+
+        [HarmonyPostfix]
+        internal static void Postfix(IngameMenu __instance)
+        {
+            __instance.quitToMainMenuText.text = $"{(GameModeUtils.IsPermadeath() ? "Save & " : "")}Quit to Main Menu";
+            IngameMenu_Start_Patch.quitButton.interactable = !GameModeUtils.IsPermadeath();
+            IngameMenu_Start_Patch.quitButton.GetComponentsInChildren<Text>().Do(t => t.text = "Quit to Desktop");
+        }
+    }
+
+    /*
+    [HarmonyPatch(typeof(IngameMenu), nameof(IngameMenu.QuitGame))]
+    internal static class IngameMenu_QuitGame
+    {
+        // This is a pass-through patch
+
+        [HarmonyPrefix]
+        internal static bool Prefix(IngameMenu __instance, bool quitToDesktop)
+        {
+            __instance.StartCoroutine(IngameMenu_QuitGameAsync.Passthrough(__instance.QuitGameAsync(quitToDesktop), quitToDesktop));
```

Comment:
If you're having to make so many changes to the original mod,  
then you evidently don't think the mod in its original form is something that everyone should and would choose to use on their own.

So here you're trying to find a variation of this mod that would be good for everyone  
but in having to go through all that, you're pretty much giving all the evidence needed to say that this mod _isn't_ something that should be merged into the basic mod loader.

You are merging in a mod, but then it is different from the original,  
so people who actually want the original QtD in it's original form, won't be able to use it as it was
and saying "they can just Alt+F4" isn't a good answer.

My review is this:  
If QtD can't be merged without all these alterations that change how it originally worked,  
then it shouldn't be merged at all.

Reply:
A Quit to Desktop button is something that all games should have. The only change I made to the mod was that I made the qtmm "are you sure" dialog also appear appear for the qtd button.

Reply:
Then you aren't merging in the same mod anymore.

If you want to have this an an option, ask Randy if you can create a variant of QtD or submit to PR to Randy's mod that makes that optional.

Either way, this isn't the same mod anymore and you are going to be removing the option for people to use old behavior from people who might prefer it.

That is why my vote is still 'no' on merging this mod.

Reply:
As an idea, I understand what you mean. In this case, though, the "old behavior" is one less click.

Reply:
> As an idea, I understand what you mean. In this case, though, the "old behavior" is one less click.

You'd be surprised what people will complain about.